src/api/eventTypes.js

```diff
@@ -149,21 +149,91 @@ type StreamListEvent = $ReadOnly<{|
   streams: $ReadOnlyArray<Stream>,
 |}>;
 
+type StreamUpdateEventBase = $ReadOnly<{|
+  ...EventCommon,
+  type: typeof EventTypes.stream,
+  op: 'update',
+  stream_id: number,
+  name: string,
+|}>;
+
+// https://zulip.com/api/get-events#stream-update
+export type StreamUpdateEvent =
+  | {| ...StreamUpdateEventBase, +property: 'name', +value: $PropertyType<Stream, 'name'> |}
+  | {|
+      ...StreamUpdateEventBase,
+      +property: 'description',
+      +value: $PropertyType<Stream, 'description'>,
+      +rendered_description: $PropertyType<Stream, 'rendered_description'>,
+    |}
+  // TODO(server-4.0): New in FL 30.
+  | {|
+      ...StreamUpdateEventBase,
+      +property: 'date_created',
+      +value: $PropertyType<Stream, 'date_created'>,
+    |}
+  | {|
+      ...StreamUpdateEventBase,
+      +property: 'invite_only',
+      +value: $PropertyType<Stream, 'invite_only'>,
+      +history_public_to_subscribers: $PropertyType<Stream, 'history_public_to_subscribers'>,
+
+      // TODO(server-5.0): New in FL 71.
+      +is_web_public?: $PropertyType<Stream, 'is_web_public'>,
+    |}
+  | {|
+      ...StreamUpdateEventBase,
+      +property: 'rendered_description',
+      +value: $PropertyType<Stream, 'rendered_description'>,
+    |}
+  | {|
+      ...StreamUpdateEventBase,
+      +property: 'is_web_public',
+      +value: $PropertyType<Stream, 'is_web_public'>,
+    |}
```

Comment:
I think these can usefully be further deduplicated. I'll send that as a quick PR on top of this.

Reply:
#5350